bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java

```diff
@@ -0,0 +1,321 @@
+     * Possible Instructions are:
+     * FssTransmit 1 Kommandoabsetzung (Controller > eGate > Dominoswiss)
+     * FssReceive 2 Empfangenes Funkpaket (Dominoswiss > eGate > Controller)
+     *
+     * @throws InterruptedException
+     *
+     */
+
+    public void tiltUp(String id) throws InterruptedException {
+        for (int i = 0; i < 3; i++) {
+            pulseUp(id);
+            Thread.sleep(150); // sleep to not confuse the blinds
+
+        }
+    }
+
+    public void tiltDown(String id) throws InterruptedException {
+        for (int i = 0; i < 3; i++) {
+            pulseDown(id);
+            Thread.sleep(150);// sleep to not confuse the blinds
+        }
+    }
+
+    public void pulseUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=1;Priority=1;CheckNr=3415347;" + CR);
+    }
+
+    public void pulseDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=2;Priority=1;CheckNr=2764516;" + CR);
+    }
+
+    public void continuousUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=3;Priority=1;CheckNr=2867016;" + CR, 20000);
+    }
+
+    public void continuousDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=4;Priority=1;CheckNr=973898;" + CR, 20000);
+    }
+
+    public void stop(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=5;Priority=1;CheckNr=5408219;" + CR);
+    }
+
+    // @SuppressWarnings("null")
+    public void registerBlind(String id, ThingUID uid) {
+        logger.debug("Registring Blind id {} with thingUID {}", id, uid);
+        registeredBlinds.put(id, uid);
+    }
+
+    /**
+     * Send a command to the eGate Server.
+     */
+
+    private void sendCommand(String command) {
+        sendCommand(command, SOCKET_TIMEOUT_SEC);
+    }
+
+    @SuppressWarnings("null")
+    private synchronized void sendCommand(String command, int timeout) {
+        logger.debug("EGate got command: {}", command);
+
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        try {
+            egateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+            writer.write(command);
+            writer.flush();
+        } catch (IOException e) {
+            logger.error("Error while sending command {} to Dominoswiss eGate Server {} ", command, e.toString());
+        }
+    }
+
+    private void pollingConfig() {
+        if (!isConnected()) {
+            synchronized (lock) {
+                try {
+                    egateSocket = new Socket();
+                    logger.debug("before connect");
+                    egateSocket.connect(new InetSocketAddress(host, port));
+                    logger.debug("after connect");
+                    egateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+                    logger.debug("before writer");
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging
```
void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```